runtime/contract_update_validation.go

```diff
@@ -457,9 +475,30 @@ func identifiersEqual(expected []ast.Identifier, found []ast.Identifier) bool {
 	}
 
 	for index, element := range found {
-		if expected[index] != element {
+		if expected[index].Identifier != element.Identifier {
 			return false
 		}
 	}
 	return true
 }
+
+func isAnyStructOrAnyResourceType(restrictedType ast.Type) bool {
```

Comment:
Given that the function works for any type, maybe generalize the parameter name from `restrictedType` to e.g. `ty`